scene/3d/ray_cast.cpp

```diff
@@ -231,6 +240,47 @@ void RayCast::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "type_mask", PROPERTY_HINT_FLAGS, "Static,Kinematic,Rigid,Character,Area"), "set_type_mask", "get_type_mask");
 }
 
+void RayCast::_create_debug_shape() {
+
+	if (debug_shape) {
+		debug_shape->queue_delete();
```

Comment:
This is apparently not enough to delete the heap allocated memory, as it is leaking quite badly. I suspect the `Mesh` and `MeshInstance` allocated with `memnew` are the culprits.